src/components/Bridge/TokenInput.tsx

```diff
@@ -15,29 +15,153 @@ import {
+import { useWalletSelector } from "components/WalletSelectorContextProvider"
+import Decimal from "decimal.js"
+import { providers } from "near-api-js"
+import { CodeResult } from "near-api-js/lib/providers/provider"
+import { useCallback, useEffect, useMemo, useState } from "react"
+import { AiFillCloseCircle } from "react-icons/ai"
+import useSWR from "swr"
+import {
+  AppchainInfoWithAnchorStatus,
+  BridgeConfig,
+  Collectible,
+  TokenAsset,
+} from "types"
+import { DecimalUtil, ZERO_DECIMAL } from "utils"
+import { getTokenBalance } from "utils/bridge"
+import { SelectTokenModal } from "views/Bridge/SelectTokenModal"
 
 export default function TokenInpput({
   chain,
   appchain,
   from,
+  appchainId,
+  onChangeAmount,
+  onChangeTokenAsset,
+  appchainApi,
 }: {
   chain: string
   appchain?: AppchainInfoWithAnchorStatus
   from: string
+  appchainId: string
+  onChangeAmount: (value: string) => void
+  onChangeTokenAsset: (value: TokenAsset) => void
+  appchainApi?: ApiPromise
 }) {
+  const { accountId, selector } = useWalletSelector()
+
   const bg = useColorModeValue("white", "#15172c")
   const grayBg = useColorModeValue("#f2f4f7", "#1e1f34")
   const [amount, setAmount] = useState("")
   const [tokenAsset, setTokenAsset] = useState<TokenAsset>()
   const [collectible, setCollectible] = useState<Collectible>()
+  const [balance, setBalance] = useState<Decimal>()
   const [isAmountInputFocused, setIsAmountInputFocused] = useBoolean()
   const [selectTokenModalOpen, setSelectTokenModalOpen] = useBoolean()
+  const [isLoadingBalance, setIsLoadingBalance] = useBoolean()
+
+  const { data: tokens } = useSWR<TokenAsset[]>(
+    appchainId ? `tokens/${appchainId}` : null
+  )
+  const { data: bridgeConfig } = useSWR<BridgeConfig>(
+    appchainId ? `bridge-config/${appchainId}` : null
+  )
+  const { data: collectibleClasses } = useSWR<number[]>(
+    appchainId ? `collectible-classes/${appchainId}` : null
+  )
+
+  const filteredTokens = useMemo(() => {
+    if (!tokens?.length) {
+      return []
+    }
+
+    if (!bridgeConfig?.whitelist) {
+      return tokens
+    }
+
+    return tokens.filter(
+      (t) =>
+        !Object.keys(bridgeConfig.whitelist).includes(t.contractId) ||
+        (Object.keys(bridgeConfig.whitelist).includes(t.contractId) &&
+          accountId &&
+          Object.values(bridgeConfig.whitelist)
+            .flat(Infinity)
+            .includes(accountId))
+    )
+  }, [tokens, bridgeConfig, accountId])
+
+  const onUpdateAmount = useCallback((v: string) => {
+    setAmount(v)
+    onChangeAmount(v)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+  const isNear = chain === "NEAR"
+  const onSetMax = () => {
+    if (!isNear && tokenAsset?.assetId === undefined) {
+      onUpdateAmount(
+        balance?.sub(0.1).gt(ZERO_DECIMAL) ? balance?.sub(0.1).toString() : ""
+      )
+    } else {
+      onUpdateAmount(balance?.toString() || "")
```

Comment:
0.1 native token for EVM appchain gas fee is too much. BaseFee in moonbeam appchin right now is a constant named gasPrice. 200,000 * 10 * gasPrice is enough, for barnacle-evm, it is only 0.0002.